### PR TITLE
Add filter pattern concept documentation

### PR DESCRIPTION
**Pull request by jorytindall**
### :pushpin: Summary

This PR adds various documentation around how to conceptualize filters and introduces some language that allows for easier communication about the specifics. This is less about tactical implementation of filters, but more how various methods of communication impact the UX and the conceptual anatomy of a filter.

### :link: External links

Jira ticket: [HDS-644](https://hashicorp.atlassian.net/browse/HDS-644)

***

### 👀 Reviewer's checklist:

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-644]: https://hashicorp.atlassian.net/browse/HDS-644?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/patterns/filter-patterns/partials/core-concepts/core-concepts.md [hunk 1]
```diff
@@ -1,47 +1,80 @@
-## Filter concepts and terminology
+Filtering consists of several conceptual foundations that contribute the larger pattern, filtering method, and communication of filters to the user.
```
Comment (alex-ju) [inline website/docs/patterns/filter-patterns/partials/core-concepts/core-concepts.md:1]:
```suggestion
Filtering consists of several conceptual foundations that contribute to the larger pattern, filtering method, and communication of filters to the user.
```

**Review by alex-ju — APPROVED** [review]
Left a few minor comments – it looks like it would need a rebase before merging it in. Let me know if you run into any issues with getting the filter-pattern branches to play nicely together.

File: website/docs/patterns/filter-patterns/partials/core-concepts/core-concepts.md [hunk 2]
```diff
@@ -1,47 +1,80 @@
...
 
-### Identifier
+## Concepts and terminology
 
-**Identifier** refers to the specific property or category that is being filtered on and is often represented by the specific label of the filter.
+| Concept | Usage |
+|---------|-------|
+| Parameter | Refers to the specific property or category that is being filtered on; often represented by the specific label of the filter. |
+| Value | Refers to the specific value within a parameter that is referenced in the filter; e.g., a specific string, status, numerical range, etc. |
+| Conditional | Refers to the relationship between the parameter and the value. |
```
[the remark this answers is not in the document]
Comment (jorytindall) [reply website/docs/patterns/filter-patterns/partials/core-concepts/core-concepts.md:9]: I noticed that as well, appears that it's when the table is followed by a paragraph (or basically a non-headline)

**Review by heatherlarsen — APPROVED** [review]
:shipit:

**Pull request merged**